### PR TITLE
Azure template fix

### PR DESCRIPTION
**Pull request by paulvanbrenk**
The NodeJs templates in VS no longer work on Azure. This cleans up the templates to they work without changes.

### CODE REVIEW

File: Nodejs/Product/Nodejs/ProjectTemplates/AzureExpress4App/app.js [hunk 1]
```diff
@@ -57,5 +57,8 @@ app.use(function (err, req, res, next) {
     });
 });
 
+app.set('port', process.env.PORT || 3000);
 
-module.exports = app;
+var server = app.listen(app.get('port'), function () {
+    debug('Express server listening on port ' + server.address().port);
```
Comment (billti) [inline Nodejs/Product/Nodejs/ProjectTemplates/AzureExpress4App/app.js]: Did you try running this and seeing if there are errors? I don't see where 'debug' is defined/imported, so I assume this line throws.

Comment (paulvanbrenk) [inline Nodejs/Product/Nodejs/ProjectTemplates/AzureExpress4App/app.js]: Forgot to comment that out in the template.

File: Nodejs/Product/Nodejs/ProjectTemplates/AzureNodejsApp/Web.config [hunk 2]
```diff
@@ -19,35 +19,37 @@
...
 
     <!-- indicates that the server.js file is a Node.js application 
     to be handled by the iisnode module -->
     <handlers>
       <add name="iisnode" path="server.js" verb="*" modules="iisnode" />
 
-      <!-- Remote debugging (Azure Website with git deploy): Uncomment NtvsDebugProxy handler below.
-      Additionally copy Microsoft.NodejsTools.WebRole to 'bin' from the Remote Debug Proxy folder.-->
-      <!--<add name="NtvsDebugProxy" path="ntvs-debug-proxy/$guid2$" verb="*" resourceType="Unspecified"
-        type="Microsoft.NodejsTools.Debugger.WebSocketProxy, Microsoft.NodejsTools.WebRole"/>-->
```
Comment (billti) [inline Nodejs/Product/Nodejs/ProjectTemplates/AzureNodejsApp/Web.config]: So still removing this bit then - not needed?

File: Nodejs/Product/Nodejs/Nodejs.csproj [hunk 3]
```diff
@@ -1293,20 +1275,15 @@
...
     </ZipProject>
     <ZipProject Include="ProjectTemplates\AzureNodejsApp\package.json" />
     <ZipProject Include="ProjectTemplates\AzureNodejsApp\README.md" />
-    <ZipProject Include="ProjectTemplates\AzureNodejsApp\setup_web.cmd" />
     <ZipProject Include="ProjectTemplates\AzureNodejsApp\Web.Debug.config">
       <SubType>Designer</SubType>
     </ZipProject>
     <ZipProject Include="ProjectTemplates\AzureNodejsApp\Web.config" />
-    <ZipProject Include="ProjectTemplates\AzureNodejsApp\node.cmd" />
-    <ZipProject Include="ProjectTemplates\AzureNodejsApp\iisnode.yml" />
```
Comment (billti) [inline Nodejs/Product/Nodejs/Nodejs.csproj]: Why are you leaving the iisnode.yml file in some templates, and not others? Is it needed in some?

File: Nodejs/Product/Nodejs/ProjectTemplates/TypeScriptAzureWebApp/package.json [hunk 4]
```diff
@@ -2,8 +2,13 @@
   "name": "$npmsafeprojectname$",
   "version": "0.0.0",
   "description": "$projectname$",
-  "main": "app.js",
+  "scripts": {
+    "start": "node server"
```
Comment (billti) [inline Nodejs/Product/Nodejs/ProjectTemplates/TypeScriptAzureWebApp/package.json]: Did the main module change from app.js to server.js?

File: Nodejs/Product/Nodejs/ProjectTemplates/TypeScriptAzureWebRole/package.json [hunk 5]
```diff
@@ -2,8 +2,13 @@
   "name": "$npmsafeprojectname$",
   "version": "0.0.0",
   "description": "$projectname$",
-  "main": "app.js",
+  "scripts": {
+    "start": "node server"
```
Comment (billti) [inline Nodejs/Product/Nodejs/ProjectTemplates/TypeScriptAzureWebRole/package.json]: likewise

File: Nodejs/Product/Nodejs/ProjectTemplates/Express4App/app.js [hunk 6]
```diff
@@ -57,5 +57,8 @@ app.use(function (err, req, res, next) {
     });
 });
 
+app.set('port', process.env.PORT || 3000);
 
-module.exports = app;
+var server = app.listen(app.get('port'), function () {
+    debug('Express server listening on port ' + server.address().port);
```
Comment (billti) [inline Nodejs/Product/Nodejs/ProjectTemplates/Express4App/app.js]: same issue

File: Nodejs/Product/Nodejs/ProjectTemplates/TypeScriptExpressApp/app.ts [hunk 7]
```diff
@@ -46,5 +46,8 @@ app.use((err: any, req, res, next) => {
     });
 });
 
+app.set('port', process.env.PORT || 3000);
 
-module.exports = app;
+var server = app.listen(app.get('port'), function () {
+    debug('Express server listening on port ' + server.address().port);
```
Comment (billti) [inline Nodejs/Product/Nodejs/ProjectTemplates/TypeScriptExpressApp/app.ts]: same issue

File: Nodejs/Product/Nodejs/Nodejs.csproj (hunk 3 above)
Comment (paulvanbrenk) [inline Nodejs/Product/Nodejs/Nodejs.csproj]: removed from the actual projects, put left in the csproj file

File: Nodejs/Product/Nodejs/ProjectTemplates/AzureNodejsApp/Web.config (hunk 2 above)
Comment (paulvanbrenk) [inline Nodejs/Product/Nodejs/ProjectTemplates/AzureNodejsApp/Web.config]: Missed in reverting change

File: Nodejs/Product/Nodejs/ProjectTemplates/TypeScriptAzureWebApp/package.json (hunk 4 above)
Comment (paulvanbrenk) [inline Nodejs/Product/Nodejs/ProjectTemplates/TypeScriptAzureWebApp/package.json]: yep

File: Nodejs/Product/Nodejs/ProjectTemplates/TypeScriptAzureWebRole/package.json (hunk 5 above)
Comment (paulvanbrenk) [inline Nodejs/Product/Nodejs/ProjectTemplates/TypeScriptAzureWebRole/package.json]: yep changed from app to server

File: Nodejs/Product/Nodejs/ProjectTemplates/Express4App/app.js (hunk 6 above)
Comment (paulvanbrenk) [inline Nodejs/Product/Nodejs/ProjectTemplates/Express4App/app.js]: fixed

File: Nodejs/Product/Nodejs/ProjectTemplates/TypeScriptExpressApp/app.ts (hunk 7 above)
Comment (paulvanbrenk) [inline Nodejs/Product/Nodejs/ProjectTemplates/TypeScriptExpressApp/app.ts]: and fixed

**Pull request merged**